pkg/fleetautoscalers/fleetautoscalers.go

```diff
@@ -404,31 +444,50 @@ func applyChainPolicy(c autoscalingv1.ChainPolicy, f *agonesv1.Fleet, gameServer
 	replicas := f.Status.Replicas
 	var limited bool
 	var err error
+	var chainEntry autoscalingv1.FleetAutoscalerPolicyType
 
 	// Loop over all entries in the chain
 	for _, entry := range c {
```

Comment:
The applyChainPolicy iterates through the policies in the chain. If an error occurs for a specific policy, the loop continues. Only if all policies in the chain fail will the function return an error. The logging and event emission in `applyChainPolicy` happen only if an error occurs or if there's a change in the applied policy. Is this interpretation correct @indexjoseph I might have forgotten.